lib/application_insights/channel/contracts/data.rb

```diff
@@ -1,48 +1,9 @@
 require_relative 'json_serializable'
 
-module ApplicationInsights
-  module Channel
-    module Contracts
-      # Data contract class for type Data.
-      class Data < JsonSerializable
-        # Initializes a new instance of the Data class.
-        def initialize(options={})
-          defaults = {
-            'baseType' => nil,
-            'baseData' => nil
-          }
-          values = {
-            'baseData' => nil
-          }
-          super defaults, values, options
-        end
-        
-        # Gets the base_type property.
-        def base_type
-          @values.fetch('baseType') { 
-            @values['baseType'] = nil
-          }
-        end
-        
-        # Sets the base_type property.
-        def base_type=(value)
-          if value == @defaults['baseType']
-            @values.delete 'baseType' if @values.key? 'baseType'
-          else
-            @values['baseType'] = value
-          end
-        end
-        
-        # Gets the base_data property.
-        def base_data
-          @values['baseData']
-        end
-        
-        # Sets the base_data property.
-        def base_data=(value)
-          @values['baseData'] = value
-        end
-      end
-    end
+module ApplicationInsights::Channel::Contracts
+  class Data
+    include JsonSerializable
+
+    attributes :baseType, :baseData
```

Comment:
Something that I noticed when I was implementing **Example 4** is that we don't keep a list of the attributes, something necessary for the JSON serialization. We can get them from two places:

1. `instance_variables` Not recommendable because obvious reasons.
2. `json_mapping.keys` But we need to assert that `attr_accessor` defines the same keys than this hash and also requires that all the keys **must** define a mapping string even when they are the same.

### Redundant mapping definition.
```ruby
    attr_accessor :name, :kind, :value, :count, :min, :max, :std_dev

    json_mapping name: 'name', kind: 'kind', value: 'value', count: 'count', min: 'min', max: 'max', std_dev: 'stdDev'
```